SU-HSDO/suhumsci
# STN-692 : Create Vertical Timeline Component

# [STN-692](https://sparkbox.atlassian.net/browse/STN-692) READY FOR REVIEW

## Summary
The work in this PR creates a Vertical Timeline component that allows for multiple “timeline item” elements to be styled together so that presents a step-by-step content flow in a clear format to users.

_Note:_ The vertical timelines is not currently scoped to work with dark color inversion.

## Need Review By (Date)
asap

## Urgency
hot 🔥 

## Steps to Test
1. In your CLI, run `npm run test` and confirm all tests pass.
2. In order for the vertical timeline and timeline patterns to be recognized, you will need to clear your cache by running `lando drush @sparkbox_sandbox.local cr`. Run `lando drush @sparkbox_sandbox.local config-import --partial` to import the config files.
3. To test a vertical timeline, create a new flexible page or open an existing one. Under Components add a Vertical Timeline. Fill in all the fields.

Test the following scenarios:
- [x] A vertical timeline with the Add Expand/Collapse checkbox NOT checked. The timeline should be expanded by default on page load. When the timeline is expanded, there should not be an arrow icon to the right of the timeline items.
- [x] A vertical timeline with the Add Expand/Collapse checkbox checked. The timeline should be collapsed. There should be an arrow icon visible to the right of the timeline item.
- [x] A vertical timeline with a single timeline item should only have the circle icon to the left. It should not have a vertical pipe connected to it.
- [x] A vertical timeline in a row.
- [x] A vertical timeline in a well.
- [x] A vertical timeline in a row beside another component such as a text area.
- [x] A vertical timeline with a timeline item that only has a summary. Confirm this is the only required field.
- [x] A vertical timeline with a timeline item that has a summary and subheading only.
- [x] A vertical timeline with a timeline item that has a summary, subheading, and description.
- [x] Test both Traditional and Colorful themes. The only difference you should see is the line-height of the description. In Colorful the line height is a little more generous.

4. Confirm the timeline item pattern is not available as a paragraph in a row or in the components field on a flexible page.
